### ISSUE TITLE
Output Files location and readme

### ISSUE THREAD

**Issue by npatel22526** [mehrdadbakhtiari/adVNTR#17]
Hello, 

I was trying to test the adVNTR tools, it installed without any issue and I obtained vntr_data following  instructions in docs. Below is my command that I ran 
advntr genotype --alignment_file Bams/Samples.bam --working_directory Bams/ 

I don't seem to find any information about the output file. Can you please provide some information on where to find output file as well brief description to understand the output file. 

Thanks, 
Nick

**Comment by mehrdadbakhtiari** [comment]
Hi Nick,

I am sorry that this information was not clear in the documentation, adVNTR writes the results in stdout. I am going to keep this issue open until I add the option to specify the output file.

With regard to your second question, it writes two lines per locus. First line is the ID associated with the VNTR (repeating region), and the second line contains the diploid copy numbers ("R1/R2" where R1 and R2 are repeating numbers). This information is somehow missed in moving the documentation from README to readthedocs, I'm going to add this soon. Thanks for catching it.

I'd be happy to answer any other questions that you may have.
Mehrdad

**Comment by npatel22526** [comment]
Hello Mehrdad, 

Thanks for the information. I will review the results and  get back to you if I have questions. 
A very useful feature will be to output the results along with the reference information, because in order to match ID's one has to first go back to view model.
An example output could be something like below,  (number's aren't reals)
#CHROM	Start	END	VNTR_ID	Gene	Motif RefCopy	R1	R2
chr12	117799644	117799884	600000	NOS1	CA	20	20	25

Couple questions 
Also why there are  two bp motifs in "vntr_data/hg19_VNTRs.db"  ?
What are the *json output files in vntr_data ?

Thanks in advance for your help. 
Nick

**Comment by mehrdadbakhtiari** [comment]
Yes that is a great suggestion and it is easy to implement. In addition to that, I'm working on generating VCF file which has comprehensive information about the repeat and its much more convenient to use.

json files store the HMM structure (used to count the repeats in PacBio sequencing data, and select Illumina reads and count repeats in Illumina reads). They are not output files but rather intermediate results or part the process.

I am not sure whether I understood your question about hg19_VNTRs.db. Are you referring to the motif in NOS1 gene (CA pattern)? I believe that's the only dinucleotide pattern there and we used it to see the performance of our method on STRs (which have much shorter patterns).

Best,
Mehrdad

**Comment by npatel22526** [comment]
It's good to hear that it is easy implementation. Yes, VCF would certainly be more useful but quick bed files can do the jobs most of the time. Hopefully updated version is coming soon. 

Thanks, Understood about json output. 

Yes, I was referring to the dinucleotide motif in NOS1 (ID 6000) and the reason I asked is because in my test analysis there were only 4 IDs in the output file (Is it expected in 30X in illumina WGS ? ) and one of them was the ID 6000 in NOS1 , so I was surprise to see dinucleotide in my results. Should I just ignore them ? Or I should generate model database from  UCSC simple repeat track.

Thanks again for your quick response. 
Nick

**Comment by mehrdadbakhtiari** [comment]
Hi Nick,

Thanks for you suggestion. I added a parameter to get BED output file. Please download the latest version 1.2.0 (preferably through conda) and use `--outfmt bed` in you command to see the output in the format that you suggested. I had to include other commits in this version which include some major changes that mostly speed up the genotyping. All the tests run successfully and results seem fine and I'm continuing to tests this version more in the following days, however, you may find some errors and I would appreciate it if you let me know if this is the case.

Please also re-download vntr_data. With the new code and new database, you get more comprehensive list of VNTRs in the output by default.

You can ignore some cases such as NOS1, or if you are interested in a specific gene you can extract the regions with `advntr viewmodel --gene GENE` and give ids to `advntr genotype --vntr_id`

Best,
Mehrdad

**Comment by mehrdadbakhtiari** [comment]
Commit https://github.com/mehrdadbakhtiari/adVNTR/commit/c59ad2ffdf8311919b0438c8c3941a335560eb20 added `--outfile` arg to specify the output file. This commit will be included in the next release.
Also, description of the outputs are now available in the [outputs](https://advntr.readthedocs.io/en/latest/tutorial.html#outputs) section of the documentations. 

Please let me know whether you need further help.
Thanks

**Comment by npatel22526** [comment]
Hello Mehrdad, 

Thank you very much for implementing the changes. To give you an update , with the new changes (1.2.0) I am getting bed files as stated above. Don't have exact numbers as it takes quite a time to finish a WGS samples but output seems to have decent number asVNTRs. 

Thanks, 
Nick

**Comment by npatel22526** [comment]
Hello Mehrdad,

Opening it again with two questions. 
1) How adVNTR assigns diploid copy number ? 
2) Each of my files outputs exactly contains 6720 lines, The number sounds too small compared to 50-100,000 VNTRs in genome.

Thanks,
Nick

**Comment by npatel22526** [comment]
Hello Mehrdad, 

Thank you very much for your detail response for both the questions. 

 I do have few suggestions to make, 
- First and foremost, it would be very helpful to pointing out the fact that only 7000 loci will be generated in both Readme as well as in manual. 
- I like the idea of creating two separate DBs but the tool manual says that the default model is vntr_data/hg19_VNTRs.db. When I checked the model file before running the adVNTR it contained ~156K loci , so not sure where the alleles were filtered out ? 
>  -m/--models <file>          file containing VNTRs information [vntr_data/hg19_VNTRs.db]  
- I do understand that the running time will drastically be increased but one can split the input file into pieces and run it in parallel and then it's just a matter of combing bed files. 

Yes, there is an option to provide vntr_id but "comma-separated list of VNTR IDs" is not very ideal for parallel processing   , would it be possible to read the IDs from files. 

Sincerely appreciate your responses
Nick

**Comment by mehrdadbakhtiari** [comment]
Hi Nick,

I made the changes that you suggested to both Readme and manual.

I also made separate databases that include either ~7000 or ~150K loci and removed any filtering strategy in the code. So you'll be able to genotype all VNTRs in the database that you donwload.

Yes, your suggestion about splitting the file and parallel execution makes perfect sense.

I will close this issue and create another issue about reading VNTR IDs from a file instead of writing them in command line. Since there are separate databases and there is possibility of having multiple smaller files, I guess it will have lower priority but I'll try to add this feature soon.

Thank you for your help and suggestions,
Mehrdad
